PiETLab/TreasureBoxBrailleApps
# Fix/logging user action

# Description

What:
I extended the existing logging code to include all authoring app actions. The logging file appends user actions and has a max file size.

Why:
A previous issue asked for detailed logging. We now have more detailed information about what actions the users take.

How:
Utilized a different constructor for the FileHandler to add appending to the file, and changed the file limit from default to 1MB
Leveraged the logger from the GUI in InsertBefore and RightPanel classes to add logging. 

Time:
Less than 15 minutes to read and review the code and PR.
Approximately 15 minutes to test code changes. 

## Added

- Added a constant value for the max log file size in GUI class.
- Added boolean value "true" to FileHandler constructor for appending
- Added logger lines into InsertBefore and RightPanel Class 

## Changed

- Changed to different FileHandler constructor to use file appending
- Changed max file value from default max to 1MB with constant value


# How should this be manually tested?

Checking Max File Size:
1. Change 'fileSizeLimit" to 200
2. Run Authoring app
3. Select "New Scenario"
4. Select "New Item"
5. Check log for added items
6. Select "New Item" x2
7. Check if log file has been reset.
8. Change "fileSizeLimit back to "1024000"

Checking All user actions are logged:
1. Run authoring app
2. Select "New Scenario"
3. Select all available user options from "New Item" to "Test Scenario"
4. Check if log file contains selected user actions
